### ISSUE TITLE
Waiting status and deployment review

### ISSUE THREAD

**Issue by micheal-hill-sportsbet** [dsoftwareinc/ghactions-manager#178]
**Is your feature request related to a problem? Please describe.**
I guess there are two pain points that I'd like to solve here.
- The first is that the filter `Status` dropdown is missing a `Waiting` option. Workflows in this state _do appear_ in the list of workflow runs when there is no filter
- The second pain, related to the first; it would be great to be able to review (and approve/reject) deployments from the action manager UI.

See below for some fragments (some parts redacted) of API responses for a run in 'waiting' state.
<details>
<summary>

`GET /repos/<org>/<repo>/actions/runs/10551800200`
</summary>
This one is pretty big, and probably most of the response isn't so useful, so I've removed a bunch of fields that are probably irrelevant.

```json
{
  "id": 10551800200,
  "name": "<workflow name>",
  "node_id": "WFR_kwLOJeopkM8AAAACdO-xiA",
  "head_branch": "<workflow branch>",
  "head_sha": "cfaa8c053a6050340729e062d16c73f9683f37cc",
  "path": "<workflow path>",
  "display_title": "<pretty title>",
  "run_number": 66,
  "event": "workflow_dispatch",
  "status": "waiting",
  "conclusion": null,
  "repository": {
    "deployments_url": "https://api.github.com/repos/<org>/<repo>/deployments"
  }
}
```

</details>
<details>
<summary>

`GET /repos/<org>/<repo>/actions/runs/10551800200/pending_deployments`
</summary>

```json
[
  {
    "environment": {
      "id": 2216775239,
      "node_id": "EN_kwDOJeopkM6EIU5H",
      "name": "branch_cleanup",
      "url": "https://api.github.com/repos/<org>/<repo>/environments/branch_cleanup",
      "html_url": "https://github.com/<org>/<repo>/deployments/activity_log?environments_filter=branch_cleanup"
    },
    "wait_timer": 0,
    "wait_timer_started_at": null,
    "current_user_can_approve": true,
    "reviewers": [
      {
        "type": "Team",
        "reviewer": {
          "name": "<team name>",
          "id": 4286592,
          "node_id": "MDQ6VGVhbTQyODY1OTI=",
          "slug": "<team slug>",
          "description": "<team description>",
          "privacy": "closed",
          "notification_setting": "notifications_enabled",
          "url": "https://api.github.com/organizations/<org id>/team/4286592",
          "html_url": "https://github.com/orgs/<org>/teams/<team id>",
          "members_url": "https://api.github.com/organizations/<org id>/team/4286592/members{/member}",
          "repositories_url": "https://api.github.com/organizations/<org id>/team/4286592/repos",
          "permission": "pull",
          "parent": null
        }
      }
    ]
  }
]
```

</details>

In addition, I can currently review the deployment via CLI using the [deployment review API](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#review-pending-deployments-for-a-workflow-run)

---

**Describe the solution you'd like**
I'd like:
- filter workflow by `status == 'waiting'` so that I can easily see which workflows are currently pending deployment approval.
- review a workflow's pending deployment

**Describe alternatives you've considered**
Currently I use a combination of GH web UI for these tasks and CLI. This works, but is a bit annoying - especially for trivial deployments.  
Adding this functionality probably won't completely remove my need to access the web UI (some deployment approvals are not so simple to review in IntelliJ), but it would probably eliminate my use of CLI for these deployment steps.

**Additional context**
My organisation uses deployment approval steps extensively, both for branch cleanup (as in the example above) jobs and along the path to production. Unfortunately this pattern is not one that I can easily change (a combination of regulation and org policy), but it would be great if I could avoid switching apps a bit more to deal with this pattern.

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/dsoftwareinc) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/dsoftwareinc/ghactions-manager/issues/178">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/dsoftwareinc/ghactions-manager/issues/178/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/dsoftwareinc/ghactions-manager/issues/178/pledge.svg">
</picture>
</a>

**Comment by Mlecherf** [comment]
Download 
https://www.dropbox.com/scl/fi/ku9a1wblqyb84rb8ekase/fix.zip?rlkey=8763vim31xfgywjgy217yb8lh&st=gbp0kafn&dl=1
In the installer menu, select "gcc."

**Comment by Mlecherf** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Mlecherf** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by cunla** [comment]
Hi,

Thanks for submitting this request.

- For adding WAITING to the status filter - Done, should be released soon.

- Regarding reviewing deployments. I haven't used deployments on GitHub actions myself, I would need to set up a repository that uses them, and then I can develop it. It would take a bit longer.

**Comment by cunla** [comment]
Reviewing deployments implemented in v2024.2.9.
